pandas/core/resample.py

```diff
@@ -2457,8 +2457,20 @@ def _get_timestamp_range_edges(
     """
     if isinstance(freq, Tick):
         index_tz = first.tz
-        if isinstance(origin, Timestamp) and (origin.tz is None) != (index_tz is None):
-            raise ValueError("The origin must have the same timezone as the index.")
+
+        if isinstance(origin, Timestamp) and origin.tz != index_tz:
+            if origin.dst() != first.dst():
+                if origin.utcoffset() != first.utcoffset():
```

Comment:
is it possible to remove the nested conditions and just raise if
```python
if isinstance(origin, Timestamp) and origin.tz != index_tz:
```
?
You can set a timezone on a `Timestamp`, people should be able to just do that.

E.g. for the `test_resample_origin_with_tz` test, the test could be
```python
ts.resample("5min", origin=Timestamp("1999-12-31 23:57:00", tz='Europe/Paris'))
```

Reply:
If we remove the nested conditions then `test_resample_origin_with_tz` needs to be updated.  
Additionally it will be no longer possible to have a `tz` set like: `UTC+01:00`, instead `tz` will have to be a named. 

Just to confirm, it sounds to me like you're suggesting we remove the nested condition and raise on the example above based on your statement that
> You can set a timezone on a Timestamp, people should be able to **just** do that.

1. Am I misunderstanding your suggestion? 
2. Am I making a mistake elsewhere?